infrastructure/resource_group_level/modules/webapps/dtfs-central-api.bicep

```diff
@@ -0,0 +1,105 @@
+  DOCKER_REGISTRY_SERVER_PASSWORD: 'test-value'
+}
+
+var dockerImageName = '${containerRegistryName}.azurecr.io/${resourceNameFragment}:${environment}'
+var dockerRegistryServerUsername = 'tfs${environment}'
+
+// https://learn.microsoft.com/en-us/azure/virtual-network/what-is-ip-address-168-63-129-16
+var azureDnsServerIp = '168.63.129.16'
+
+// These values are hardcoded in the CLI scripts
+var settings = {
+  WEBSITE_DNS_SERVER: azureDnsServerIp
+  WEBSITE_VNET_ROUTE_ALL: '1'
+  PORT: '5000'
+  WEBSITES_PORT: '5000'
+}
+
+// These values are taken from an export of Configuration on Dev (& validating with staging).
+var additionalSettings = {
+  // Note that the `webapp` module will add:
+  // APPLICATIONINSIGHTS_CONNECTION_STRING: applicationInsights.properties.ConnectionString
+  // Note that the config has APPINSIGHTS_INSTRUMENTATIONKEY as a slot setting, though the advice
+  // is not to use at the same time as its replacement - APPLICATIONINSIGHTS_CONNECTION_STRING
+  // APPINSIGHTS_INSTRUMENTATIONKEY: applicationInsights.properties.InstrumentationKey
+  DOCKER_ENABLE_CI: 'true'
+  DOCKER_REGISTRY_SERVER_URL: '${containerRegistryName}.azurecr.io'
+  DOCKER_REGISTRY_SERVER_USERNAME: dockerRegistryServerUsername
+  LOG4J_FORMAT_MSG_NO_LOOKUPS: 'true'
+  TZ: 'Europe/London'
+  WEBSITE_HTTPLOGGING_RETENTION_DAYS: '3'
+  WEBSITES_ENABLE_APP_SERVICE_STORAGE: 'false'
+}
+
+var nodeEnv = nodeDeveloperMode ? { NODE_ENV: 'development' } : {}
+
+var appSettings = union(settings, secureSettings, additionalSettings, additionalSecureSettings, nodeEnv)
+
+// These have come from the CLI scripts
+var mongoDbConnectionString = replace(cosmosDbAccount.listConnectionStrings().connectionStrings[0].connectionString, '&replicaSet=globaldb', '')
+// Note that in the CLI script, http was used, but the value in the exported config was https.
+var externalApiUrl = 'https://${externalApiHostname}'
+var connectionStrings = {
+  EXTERNAL_API_URL: {
+    type: 'Custom'
+    value: externalApiUrl
+  }
+  MONGO_INITDB_DATABASE: {
+    type: 'Custom'
+    value: cosmosDbDatabaseName
+  }
+  MONGODB_URI: {
+    type: 'Custom'
+    value: mongoDbConnectionString
+  }
+}
+
+resource cosmosDbAccount 'Microsoft.DocumentDB/databaseAccounts@2023-04-15' existing = {
+  name: cosmosDbAccountName
+}
+
+module dtfsCentralApi 'webapp.bicep' = {
+  name: 'dtfsCentralApi'
+  params: {
+    appServicePlanEgressSubnetId: appServicePlanEgressSubnetId
+    appServicePlanId: appServicePlanId
+    appSettings: appSettings
+    azureWebsitesDnsZoneId: azureWebsitesDnsZoneId
+    connectionStrings: connectionStrings
+    deployApplicationInsights: true
+    dockerImageName: dockerImageName
+    environment: environment
+    ftpsState: 'Disabled'
+    location: location
+    logAnalyticsWorkspaceId: logAnalyticsWorkspaceId
+    privateEndpointsSubnetId: privateEndpointsSubnetId
+    resourceNameFragment: resourceNameFragment
+    scmMinTlsVersion: '1.0'
```

Comment:
Note that I've raised a card to enforce 1.2, but I don't think it is relevant. `scmMinTlsVersion` is only `1.0` for the cases where `ftpsState` is disabled. I suspect we should be setting all the webapps to have `ftpsState: 'Disabled'`. That is an early enhancement.

Reply:
yep. it can be a variable as part of the repository settings - that way we can change it at any point from there